nifi-framework-api/src/main/java/org/apache/nifi/nar/NarPersistenceInfo.java

```diff
@@ -0,0 +1,60 @@
+/*
+ *  Licensed to the Apache Software Foundation (ASF) under one or more
+ *  contributor license agreements.  See the NOTICE file distributed with
+ *  this work for additional information regarding copyright ownership.
+ *  The ASF licenses this file to You under the Apache License, Version 2.0
+ *  (the "License"); you may not use this file except in compliance with
+ *  the License.  You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *  Unless required by applicable law or agreed to in writing, software
+ *  distributed under the License is distributed on an "AS IS" BASIS,
+ *  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *  See the License for the specific language governing permissions and
+ *  limitations under the License.
+ */
+
+package org.apache.nifi.nar;
+
+import java.io.File;
+import java.util.Objects;
+
+/**
+ * The information about a NAR persisted by a {@link NarPersistenceProvider}.
+ */
+public class NarPersistenceInfo {
+
+    private final File narFile;
```

Comment:
What do you think about using `Path` instead of `File` for a bit more abstraction of this member variable? Although they are mostly interchangeable, `Path` provides a bit more generalization in terms of methods available on the object.

Reply:
I can update it, but it seems like not much benefit because most of the existing NAR code already works off `File`, so this new code will be passing around `Path` just to then call `toFile()` to be able to call stuff like `NarLoadResult load(Collection<File> narFiles)`

Reply:
That's a good point given the other references. Thanks.